app/src/androidTest/java/com/github/se/cyrcle/ui/map/MapScreenTest.kt

```diff
@@ -73,45 +80,30 @@ class MapScreenTest {
   @Test
   fun testMapIsDisplayed() {
     composeTestRule.setContent {
-      MapScreen(mockNavigation, parkingViewModel, userViewModel, mapViewModel, permissionHandler)
+      MapScreen(
+          mockNavigation,
+          parkingViewModel,
+          userViewModel,
+          mapViewModel,
+          permissionHandler,
+          addressViewModel)
     }
 
     composeTestRule.onNodeWithTag("MapScreen").assertIsDisplayed()
     composeTestRule.onNodeWithTag("NavigationBar").assertIsDisplayed()
-
-    // Assert that the zoom controls are displayed
-    composeTestRule.onNodeWithTag("ZoomControlsIn").assertIsDisplayed().assertHasClickAction()
-    composeTestRule.onNodeWithTag("ZoomControlsOut").assertIsDisplayed().assertHasClickAction()
-  }
-
-  /**
-   * Test to verify that the internal state of the zoom level decreases correctly.
-   *
-   * This test sets the content to the `MapScreen` with an initial zoom level state, performs zoom
-   * out actions, and asserts that the state value matches the minimum zoom level.
-   */
-  @Test
-  fun testInternalStateOut() {
-
-    val state = mutableStateOf(defaultZoom)
-
-    composeTestRule.setContent {
-      MapScreen(
-          mockNavigation, parkingViewModel, userViewModel, mapViewModel, permissionHandler, state)
-    }
-
-    for (i in 0..(defaultZoom - minZoom).toInt()) {
-      composeTestRule.onNodeWithTag("ZoomControlsOut").performClick()
-    }
-
-    // Assert the state value
-    assert(state.value == minZoom)
+    composeTestRule.onNodeWithTag("SearchBar").assertIsDisplayed()
   }
 
   @Test
   fun testAddParkingRules() {
     composeTestRule.setContent {
-      MapScreen(mockNavigation, parkingViewModel, userViewModel, mapViewModel, permissionHandler)
+      MapScreen(
+          mockNavigation,
+          parkingViewModel,
+          userViewModel,
+          mapViewModel,
+          permissionHandler,
```

Comment:
Nitpick, but putting the permission handler in the middle of the ViewModels may make things more confusing. The way it was ordered before was more intuitive

Reply:
I can change it but I would like to have @andrepeiry input on this as well if possible.

Reply:
It is indeed something that would be nice to have "in-order", but it is not important to my eyes, as this is only done once at screen instantiation, so the maintenance impact isn't that big to my eyes, I feel like we can leave this as it is.

Reply:
Fine by me too then